aten/src/ATen/native/vulkan/glsl/hardsigmoid.glsl

```diff
@@ -0,0 +1,23 @@
+#version 450 core
+#define PRECISION $precision
+
+layout(std430) buffer;
+
+/* Qualifiers: layout - storage - precision - memory */
+
+layout(set = 0, binding = 0) uniform PRECISION restrict writeonly image3D   uOutput;
+layout(set = 0, binding = 1) uniform PRECISION                    sampler3D uInput;
+layout(set = 0, binding = 2) uniform PRECISION restrict           Block {
+  ivec4 size;
+} uBlock;
+
+layout(local_size_x_id = 0, local_size_y_id = 1, local_size_z_id = 2) in;
+
+void main() {
+  const ivec3 pos = ivec3(gl_GlobalInvocationID);
+
+  if (all(lessThan(pos, uBlock.size.xyz))) {
+    const vec4 outval = texelFetch(uInput, pos, 0)/6.0 + 0.5;
```

Comment:
It is useful to form the habit to add the f suffix to numeric literals to indicate a single precision floating point designation (6.0f vs 6.0) as opposed to double precision since floating point literals are double precision by default.  It is not going to make a difference here but sometimes that is the difference between calling a function that has both the single and double precision overloads (say, if a div function had both float and double overloads), or evaluating a statement at the higher double precision when single was enough.